crates/ruff_linter/src/rules/ruff/rules/chmod_non_octal.rs

```diff
@@ -0,0 +1,177 @@
+
+use crate::checkers::ast::Checker;
+use crate::{FixAvailability, Violation};
+
+/// ## What it does
+/// Checks for `chmod` calls which use non-octal integer literals.
+///
+/// ## Why is this bad?
+///
+/// Numeric modes are made up of one to four octal digits. Converting a non-octal
+/// integer to octal may not be the mode the author intended.
+///
+/// ## Example
+///
+/// ```python
+/// os.chmod("foo", 644)
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// os.chmod("foo", 0o644)
+/// ```
+///
+/// ## Fix safety
+///
+/// This rule's fix is marked as unsafe because it changes runtime behavior.
+///
+/// ## Fix availability
+///
+/// A fix is only available if the existing digits could make up a valid octal literal.
+#[derive(ViolationMetadata)]
+pub(crate) struct ChmodNonOctal {
+    reason: Reason,
+}
+
+#[derive(Debug, Clone, Copy)]
+enum Reason {
+    Decimal { found: u16, suggested: u16 },
+    Invalid,
+}
+
+impl Violation for ChmodNonOctal {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        match self.reason {
+            Reason::Decimal { found, suggested } => {
+                format!("Non-octal mode `{found}`, did you mean `{suggested:#o}`?")
+            }
+            Reason::Invalid => "Non-octal mode".to_string(),
+        }
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Replace with octal literal".to_string())
+    }
+}
+
+/// RUF064
+pub(crate) fn chmod_non_octal(checker: &Checker, call: &ExprCall) {
+    let mode_arg_index = if is_os_chmod(&call.func, checker.semantic()) {
+        1
+    } else if is_pathlib_chmod(&call.func, checker.semantic()) {
+        0
+    } else {
+        return;
+    };
+
+    let Some(mode_arg) = call.arguments.find_argument_value("mode", mode_arg_index) else {
+        return;
+    };
+
+    let Expr::NumberLiteral(ast::ExprNumberLiteral {
+        value: ast::Number::Int(int),
+        ..
+    }) = mode_arg
+    else {
+        return;
+    };
+
+    let mode_literal = &checker.locator().contents()[mode_arg.range()];
+
+    if mode_literal.starts_with("0o") || mode_literal.starts_with("0O") || mode_literal == "0" {
+        return;
+    }
+    let mode = int.as_u16();
+
+    let reason = match (mode_literal.starts_with('0'), mode) {
+        (true, _) => Reason::Invalid,
+        (false, Some(found)) => match u16::from_str_radix(&found.to_string(), 8) {
+            Ok(suggested) if suggested <= 0o7777 => Reason::Decimal { found, suggested },
+            _ => Reason::Invalid,
+        },
+        _ => Reason::Invalid,
+    };
+
+    let mut diagnostic = checker.report_diagnostic(ChmodNonOctal { reason }, mode_arg.range());
+    if let Reason::Decimal { suggested, .. } = reason {
+        let edit = Edit::range_replacement(format!("{suggested:#o}"), mode_arg.range());
+        diagnostic.set_fix(Fix::applicable_edit(edit, Applicability::Unsafe));
```

Comment:
I don't think we should offer a fix here. The main promise of the rule is that the code is probably incorrect and just rewriting the code from decimal to octal doesn't fix the root problem -> That the code is probably just incorrect.


I'd be okay if we allowlisted some common chmods and offered rewriting them but I don't think we should offer rewriting in general.